generate_latest_json.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+"""
+Sopel update-check JSON generation utility
+
+This script creates a `latest.json` file representing the most recent stable and
+unstable `sopel` releases published to PyPI, using PyPI's own JSON API.
+
+Copyright 2024 dgw, technobabbl.es
+Licensed under the Eiffel Forum License 2.
+
+https://sopel.chat
+"""
+import json
+
+from packaging.version import parse as parse_version
+import requests
+
+
+def main(argv=None):
+    data = requests.get('https://pypi.org/pypi/sopel/json').json()
```

Comment:
Let's make the error a little more obvious if something goes wrong with the request

```suggestion
    response = requests.get('https://pypi.org/pypi/sopel/json')
    response.raise_for_status()
    data = response.json()
```